File: packages/ts-rust-bridge-codegen/examples/basic.gen.rs

use bincode;
use serde::Deserialize;


#[derive(Deserialize, Debug, Clone)]
#[serde(tag = "tag", content = "value")]
pub enum Message {
    Unit,
    One(f32),
    Two(Option<bool>, f32),
    VStruct { id: String, data: String },
}


#[derive(Deserialize, Debug, Clone)]
pub struct NormalStruct {
    pub a: f32,
    pub tuple: Tuple,
}


#[derive(Deserialize, Debug, Clone)]
pub enum Enum {
    ONE,
    TWO,
    THREE,
}


#[derive(Deserialize, Debug, Clone)]
pub struct Tuple(pub Option<bool>, pub Vec<String>);


pub type Aha = Vec<Option<Vec<String>>>;

File: packages/ts-rust-bridge-codegen/examples/basic.gen.ts

export type Message = 
  | { tag: "Unit"}
  | { tag: "One", value: number}
  | { tag: "Two", value: [(boolean) | undefined, number]}
  | { tag: "VStruct", value: MessageVStruct}


export interface MessageVStruct {
      id: string;
  data: string
    }
    

export module Message {


export const Unit: Message = { tag: "Unit"};

export  const One = (value: number): Message => ({ tag: "One", value});

export  const Two = (p0: (boolean) | undefined, p1: number): Message => ({ tag: "Two", value: [p0,p1]});

export  const VStruct = (value: MessageVStruct): Message => ({ tag: "VStruct", value});


}

export interface NormalStruct {
      a: number;
  tuple: Tuple
    }
    

export enum Enum {
  ONE = "ONE",
  TWO = "TWO",
  THREE = "THREE"
}


export interface Tuple {
      0: (boolean) | undefined;
  1: Array<string>;
  length: 2
    }
    

export module Tuple {


export  const mk = (p0: (boolean) | undefined, p1: Array<string>): Tuple => [p0, p1];


}

export type Aha = Array<(Array<string>) | undefined>
